vdirsyncer/storage/ldap.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+import logging
+import ldap3
+import vobject
+
+from .base import Item, Storage
+from .. import exceptions
+
+ldap_logger = logging.getLogger(__name__)
+
+
+class LDAPStorage(Storage):
+    '''
+    :param uri: LDAP URI
+    :param search_base: search base
+    :param bind: bind dn
+    :param password: bind password
+    :param filter: filter
+    '''
+    storage_name = 'ldap'
+    read_only = True
+    fileext = '.vcf'
+    item_mimetype = 'text/vcard'
+
+    def __init__(self, uri=None, search_base=None, bind=None, password=None,
+                 filter='(&(objectCategory=person)(objectClass=user)'
+                        '(sn=*)(givenName=*))',
+                 **kwargs):
+        super(LDAPStorage, self).__init__(**kwargs)
+        self.search_base = search_base
+        self.filter = filter
+        self.server = ldap3.Server(uri)
```

Comment:
Ah, I see, you're using https://github.com/cannatag/ldap3, but I was reading the docs for https://github.com/pyldap/pyldap

Could you switch to the latter? This would make testing easier as `mockldap` appears to support `pyldap` only.